File: Stack.java

/* DO NOT CHANGE:  This file is used in evaluation */

public class Stack<Arbitrary extends Base> extends List<Arbitrary> {

        public Arbitrary pop () {
                return remove (END);
        }

        public Arbitrary push (Arbitrary element) {
                return insert (element, END);
        }

        public Arbitrary top () {
                return view (END);
        }
}

File: Stack.c
#include <iostream>
#include "Base.h"
#include "Stack.h"
using namespace std;

/* DO NOT CHANGE:  This file is used in evaluation */

Base * Stack :: Pop (void) {
	return Remove ();
}

long Stack :: Push (Base * element) {
	return Insert (element);
}

Base * Stack :: Top (void) const {
	return View ();
}
